src/main/scala/com/databricks/labs/overwatch/utils/UpgradeV2.scala

```diff
@@ -0,0 +1,405 @@
+package com.databricks.labs.overwatch.utils
+
+import com.databricks.dbutils_v1.DBUtilsHolder.dbutils
+import com.databricks.labs.overwatch.env.Workspace
+import com.databricks.labs.overwatch.pipeline.TransformFunctions._
+import com.databricks.labs.overwatch.pipeline._
+import com.databricks.labs.overwatch.utils.Helpers.{fastDrop, parOptimize}
+import com.databricks.labs.overwatch.utils.Upgrade.spark
+import org.apache.log4j.{Level, Logger}
+import org.apache.spark.sql.DataFrame
+import org.apache.spark.sql.functions.col
+import spark.implicits._
+
+import java.util.concurrent.ConcurrentHashMap
+import scala.collection.JavaConverters.mapAsScalaConcurrentMapConverter
+import scala.collection.concurrent
+import scala.collection.mutable.ArrayBuffer
+
+class UpgradeTo0610(prodWorkspace: Workspace, startStep: Int = 1, endStep: Int = 3, enableUpgradeBelowDBR104: Boolean = false) extends UpgradeHandler {
+
+
+  def this(workspace: Workspace) {
+    this(workspace, 1, 3,false)
+  }
+
+
+  override def upgrade(): DataFrame = {
+    val tempDir = s"/tmp/overwatch/upgrade0610_status__ctrl_0x111/${System.currentTimeMillis()}"
```

Comment:
this prefix should taken from `prodWorkspace.getConfig.tempWorkingDir` by default but be able to be optionally overridden in the constructor. Please default it to null string so user doesn't have to pass it in.